File: module_user/src/main/java/com/yyxnb/module_user/viewmodel/UserViewModel.java
package com.yyxnb.module_user.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.Transformations;

import com.yyxnb.common_base.base.CommonViewModel;
import com.yyxnb.common_base.bean.UserBean;
import com.yyxnb.common_base.db.AppDatabase;
import com.yyxnb.common_base.db.UserDao;

public class UserViewModel extends CommonViewModel {

    private UserDao userDao = AppDatabase.getInstance().userDao();

    public MutableLiveData<Integer> reqUserId = new MutableLiveData<>();

    public LiveData<UserBean> getUser() {
        return Transformations.switchMap(reqUserId, input -> {
            return userDao.getUser(input);
        });
    }

    public LiveData<UserBean> getUser(int userId) {
        return userDao.getUser(userId);
    }

}

File: module_user/src/main/java/com/yyxnb/module_user/viewmodel/UserViewModel.kt
package com.yyxnb.module_user.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import com.yyxnb.common_base.bean.UserBean
import com.yyxnb.common_base.db.AppDatabase.Companion.instance
import com.yyxnb.network.BaseViewModel

class UserViewModel : BaseViewModel() {

    private val userDao = instance!!.userDao()
    @JvmField
    var reqUserId = MutableLiveData<Int>()

    val user: LiveData<UserBean>
        get() = Transformations.switchMap(reqUserId) { input: Int? -> userDao!!.getUser(input!!) }

    fun getUser(userId: Int): LiveData<UserBean> {
        return userDao!!.getUser(userId)
    }
}